File: wrappers/dotnet/indy-sdk-dotnet-test/DidTests/SetEndpointForDidTest.cs
using Hyperledger.Indy.DidApi;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System.Threading.Tasks;

namespace Hyperledger.Indy.Test.SignusTests
{
    [TestClass]
    public class SetEndpointForDidTest : IndyIntegrationTestWithPoolAndSingleWallet
    {
        [TestMethod]
        public async Task TestSetEndpointForDidWorks()
        {
            await Did.SetEndpointForDidAsync(wallet, DID1, ENDPOINT, VERKEY);
        }

        [TestMethod]
        public async Task TestSetEndpointForDidWorksForReplace()
        {
            await Did.SetEndpointForDidAsync(wallet, DID1, ENDPOINT, VERKEY);
            var receivedEndpoint = await Did.GetEndpointForDidAsync(wallet, pool, DID1);
            Assert.AreEqual(ENDPOINT, receivedEndpoint.Address);
            Assert.AreEqual(VERKEY, receivedEndpoint.TransportKey);

            var newEndpoint = "10.10.10.1:9710";
            await Did.SetEndpointForDidAsync(wallet, DID1, newEndpoint, VERKEY_MY2);
            var updatedEndpoint = await Did.GetEndpointForDidAsync(wallet, pool, DID1);

            Assert.AreEqual(newEndpoint, updatedEndpoint.Address);
            Assert.AreEqual(VERKEY_MY2, updatedEndpoint.TransportKey);
        }

        [TestMethod]
        public async Task TestSetEndpointForDidWorksForInvalidDid()
        {
            var ex = await Assert.ThrowsExceptionAsync<InvalidStructureException>(() =>
               Did.SetEndpointForDidAsync(wallet, "invalid_base58string", ENDPOINT, VERKEY)
           );
        }

        [TestMethod]
        public async Task TestSetEndpointForDidWorksForInvalidTransportKey()
        {
            var ex = await Assert.ThrowsExceptionAsync<InvalidStructureException>(() =>
               Did.SetEndpointForDidAsync(wallet, DID1, ENDPOINT, INVALID_VERKEY)
           );
        }
    }
}

File: wrappers/java/src/test/java/org/hyperledger/indy/sdk/did/SetEndpointForDidTest.java
package org.hyperledger.indy.sdk.did;

import org.hyperledger.indy.sdk.IndyIntegrationTestWithPoolAndSingleWallet;
import org.hyperledger.indy.sdk.InvalidStructureException;
import org.junit.Test;

import java.util.concurrent.ExecutionException;

import static org.hamcrest.CoreMatchers.isA;


public class SetEndpointForDidTest extends IndyIntegrationTestWithPoolAndSingleWallet {

	@Test
	public void testSetEndpointForDidWorks() throws Exception {
		Did.setEndpointForDid(wallet, DID, ENDPOINT, VERKEY).get();
	}

	@Test
	public void testSetEndpointForDidWorksForInvalidTransportKey() throws Exception {
		thrown.expect(ExecutionException.class);
		thrown.expectCause(isA(InvalidStructureException.class));

		Did.setEndpointForDid(wallet, DID, ENDPOINT, INVALID_VERKEY).get();
	}
}